common/src/main/java/com/microsoft/identity/common/internal/migration/TokenCacheItemMigrationAdapter.java

```diff
@@ -135,17 +139,40 @@
     public static boolean tryFociTokenWithGivenClientId(@NonNull final BrokerOAuth2TokenCache brokerOAuth2TokenCache,
                                                         @NonNull final String clientId,
                                                         @NonNull final String redirectUri,
-                                                        @NonNull final ICacheRecord cacheRecord)
+                                                        @NonNull final ICacheRecord cacheRecord) throws IOException, ClientException {
+        return tryFociTokenWithGivenClientId(
+                brokerOAuth2TokenCache,
+                clientId, redirectUri,
+                cacheRecord.getRefreshToken(),
+                cacheRecord.getAccount()
+        );
+    }
+
+    /**
+     * Testing whether the given client ID can use the cached foci to refresh token.
+     *
+     * @param clientId    String of the given client id.
+     * @param redirectUri redirect url string of the given client id.
+     * @param accountRecord account record of request
+     * @param refreshTokenRecord refresh token record of FOCI account
+     * @return true if the given client id can use the cached foci token. False, otherwise.
+     * @throws ClientException
+     * @throws IOException
+     */
+    public static boolean tryFociTokenWithGivenClientId(@NonNull final OAuth2TokenCache brokerOAuth2TokenCache,
+                                                        @NonNull final String clientId,
+                                                        @NonNull final String redirectUri,
+                                                        @NonNull final RefreshTokenRecord refreshTokenRecord,
+                                                        @NonNull final IAccountRecord accountRecord)
             throws ClientException, IOException {
         final String methodName = ":tryFociTokenWithGivenClientId";
         final MicrosoftStsOAuth2Configuration config = new MicrosoftStsOAuth2Configuration();
 
         //Get authority url
         final Uri.Builder requestUrlBuilder = new Uri.Builder();
-        final String tenantId = cacheRecord.getAccount().getRealm();
         requestUrlBuilder.scheme("https")
-                .authority(cacheRecord.getRefreshToken().getEnvironment())
-                .appendPath(StringUtil.isEmpty(tenantId) ? ALL_ACCOUNTS_TENANT_ID : tenantId);
+                .authority(refreshTokenRecord.getEnvironment())
+                .appendPath(StringUtil.isEmpty(accountRecord.getRealm()) ? ALL_ACCOUNTS_TENANT_ID : accountRecord.getRealm());
```

Comment:
Random comment: I kinda want to write StringUtil.defaultIfEmpty(String a, String b) -> isEmpty(a) ? b : a;